packages/beacon-node/src/network/network.ts

```diff
@@ -350,4 +358,24 @@ export class Network implements INetwork {
       }
     }
   };
+
+  private async onLightclientFinalityUpdate(finalityUpdate: altair.LightClientFinalityUpdate): Promise<void> {
+    try {
+      // messages SHOULD be broadcasted after one-third of slot has transpired
+      await this.clock.waitForSlot(finalityUpdate.signatureSlot + 1 / 3);
+      return await this.gossip.publishLightClientFinalityUpdate(finalityUpdate);
```

Comment:
I believe according to spec these messages should only be published when we have a sync committee participant.

Reply:
Indeed. I should think that has been covered [here](https://github.com/ChainSafe/lodestar/pull/4365/files#diff-4f9e095020a315582142d159413da6239fbd40c87ce2acf1f400dec9f8bee79cR476) where the actual publication occurs.

This code implements the gossip validation as specified [here](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#light_client_finality_update) and there is no mention of checking sync committee participants.

Reply:
I don't see that. The link isn't working for me.

Looking for the code that that implements this part:
https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#sync-committee
Specifically this part
> full nodes with at least one validator assigned to the current sync committee at the block's slot SHOULD broadcast

Reply:
bump on this. There should be a conditional here like:
```ts
if (this.hasAttachedSyncCommitteeMember()) {
  sleep(..)
  publish(..)
}
```

Reply:
I have seen how sync committee participation is calculated in the code base, but not come across where we check if a validator attached to the node is part of the current sync committee...

I know the validator registry can be found in the state...if I were to implement this check, my approach will be to get access to the validator registry somehow and check against the current sync committee. Will this be the most optimal approach? If there are parts of the code base where similar check is already being performed, please do share.

One meta question: The full node has already processed the block, has been able to create the lc update and also confirm there is enough sync committee participation. Do you know the rationale for this extra check?

Reply:
I think the rationale for this condition is to avoid flooding the message on the network. Only max 512 nodes will initially publish the message, rather than every full-node on the network.

Look at `beacon-node/src/network/subnets/syncnetsService.ts`. You can use `SyncnetsService#getActiveSubnets()`. If the returned array is length > 0 then we can assume we have an attached sync committee member.

Reply:
this isn't resolved yet

Reply:
opsie. I forgot to push. It is pushed now.